packages/frontend/src/components/ReactHookForm/AutoComplete.tsx

```diff
@@ -84,14 +133,47 @@ const ControlledSuggest: FC<{
         string | `${string}.${string}` | `${string}.${number}`
     >;
     suggestProps?: Partial<React.ComponentProps<typeof Suggest2<Item>>>;
-}> = ({ isLoading, suggestProps, disabled, items, field, ...props }) => {
+    groupBy?: (item: Item) => string;
+}> = ({
+    isLoading,
+    suggestProps,
+    disabled,
+    items,
+    field,
+    groupBy,
+    ...props
+}) => {
     const activeItem = items.find((item) => item.value === field.value);
     const onItemSelect = useCallback(
         (item: Item) => {
             field.onChange(item.value);
         },
         [field],
     );
+
+    const renderGroupedItemList = useMemo(() => {
```

Comment:
We should use `useCallback` instead of `useMemo` since we want to memoize a function. And `items` isn't necessary for the hook dependencies.